scripts/unpack

```diff
@@ -103,9 +108,55 @@ if [ -d "$SOURCES/$1" -o -d "$PKG_DIR/sources" -o -n "$PKG_GIT_URL" ]; then
   fi
 
   if [ -z "$PKG_URL" -a -n "$PKG_GIT_URL" ]; then
-    rm -rf $BUILD/$PKG_NAME-$PKG_VERSION
+    # init GIT_ORIG_DIR
+    GIT_ORIG_DIR=""
+    # cycle through folders that match $PKG_NAME in $BUILD until we find the right one
+    for f in $BUILD/$1-* ; do
+      echo "**** debug: checking folder '$f':"
+      if [ -d $f ]; then
+        echo "**** debug: '$f' is folder"
+        if [ -f $f/.libreelec-unpack ]; then
+          echo "**** debug: '.libreelec-unpack' found"
+          source $f/.libreelec-unpack
+          echo "**** debug: STAMP_PKG_NAME=$STAMP_PKG_NAME"
+          if [ "$PKG_NAME" = "$STAMP_PKG_NAME" ]; then
+            echo "**** debug: NAME matched"
+            if [ -f $f/.git/config ]; then
+              echo "**** debug: .git/config found:"
+              cat $f/.git/config
+              GIT_MATCH=$(grep "$PKG_GIT_URL" $f/.git/config | wc -l)
+              echo "**** debug: GIT_MATCH=$GIT_MATCH"
+              if [ "$GIT_MATCH" = "1" ]; then
+                # we found the existing folder!
+                GIT_ORIG_DIR=$f
+                break
+              fi
+            fi
+          fi
+        fi
+      fi
+    done
+    if [ -d "$GIT_ORIG_DIR" -a -n "$GIT_ORIG_DIR" ]; then
+      # rename the original folder to match current version
+      mv $GIT_ORIG_DIR $BUILD/$PKG_NAME-$PKG_VERSION
+      cd $BUILD/$PKG_NAME-$PKG_VERSION
+      # clean
+      echo "**** debug: git clean -fdx"
+      git clean -fdx
+      echo "**** debug: git checkout -- ."
+      git checkout -- .
+      echo "**** debug: git checkout master"
+      git checkout master
+      # pull changes
+      echo "**** debug: git pull"
+      git pull
+    else
+      # clean
+      $SCRIPTS/clean $1
+      # clone
     git clone $PKG_GIT_URL $BUILD/$PKG_NAME-$PKG_VERSION/
     cd $BUILD/$PKG_NAME-$PKG_VERSION/
```

Comment:
Whitespace fix.

Reply:
On which line was the incorrect whitespace? I cannot see the change.

Reply:
Ah, the `git clone` and `cd ...` should be two spaces indented to make sure the whitespace is correct.

Line 157 and 158